src/resolve/resolved-dns-query.c

```diff
@@ -734,15 +734,11 @@ int dns_query_go(DnsQuery *q) {
 
         LIST_FOREACH(scopes, s, q->manager->dns_scopes) {
                 DnsScopeMatch match;
-                const char *name;
 
-                name = dns_question_first_name(dns_query_question_for_protocol(q, s->protocol));
-                if (!name)
-                        continue;
-
-                match = dns_scope_good_domain(s, q->ifindex, q->flags, name);
+                match = dns_scope_good_domain(s, q);
                 if (match < 0) {
```

Comment:
can match ever be `< 0`? looking at `dns_scope_good_domain()` it doesn't seem so

Reply:
Indeed. The conditions are dropped. Thanks.